packages/office-ui-fabric-react/src/components/Persona/Persona.tsx

```diff
@@ -55,13 +58,33 @@ const COLOR_SWATCHES_LOOKUP: PersonaInitialsColor[] = [
 
 const COLOR_SWATCHES_NUM_ENTRIES = COLOR_SWATCHES_LOOKUP.length;
 
-export class Persona extends React.Component<IPersonaProps, any> {
+export interface IPersonaState {
+  isImageLoaded?: boolean;
+}
+
+export class Persona extends React.Component<IPersonaProps, IPersonaState> {
   public static defaultProps: IPersonaProps = {
     primaryText: '',
     size: PersonaSize.regular,
     presence: PersonaPresence.none
   };
 
+  constructor(props: IPersonaProps) {
+    super(props);
+
+    this.state = {
+      isImageLoaded: false,
+    };
+  }
+
+  public componentWillReceiveProps(nextProps: IPersonaProps) {
+    if (nextProps.imageUrl !== this.props.imageUrl) {
```

Comment:
I think you can get rid of this if you rework the event to just fire whenever the image state changes. The callback will be called as soon as the new src is provided (or not call it if the src doesn't change).